### DISCUSSION TITLE
Vim 8 ttimeout and ttimeoutlen handling?

### DISCUSSION THREAD

**Discussion by studog-github** [tpope/vim-fugitive#1871] [General]
I just updated #1745 with the results of my debugging fugitive not working on my brand new Linux Mint laptop. I had of course immediately set up my own .vimrc which caused $VIMRUNTIME/defaults.vim to be skipped (which I didn't even know about) which caused Vim's internal default values of nottimeout and ttimeoutlen=-1 to be used, which broke fugitive.

Would you be amenable to a patch to s:VersionCheck() that checks ttimeout/len and issues an error if they're not set?

**Comment by tpope** [comment]
As mentioned in #1745, I think you've misdiagnosed the problem. So no, not without further evidence.

**Comment by studog-github** [comment]
Ah crud. You are correct, I misdiagnosed the problem.

Further debugging has shown that it's same mechanism vis a vis Vim 8's defaults.vim, but it's the lack of `filetype plugin indent on` that causes the issue. Again, same, it must have been a default in the past and now gets skipped. Fix is the same, source defaults.vim, or handle manually.

Which means my question is the same, just for `filetype plugin` being enabled instead.

**Comment by studog-github** [comment]
The failure mode in fugitive is that `:G blame` works to open the commit list, but then `o` fails with the `E21` error trying to open the commit. fugitive is heisenworking which is confusing.

**Reply by tpope** [reply]
This came up recently in #1864, too. I think I'm going to attempt making those maps work without `filetype plugin on`.

**Reply by tpope** [reply]
Done.

**Reply by studog-github** [reply]
Confirmed working. Many thanks!
